Clicker/ViewControllers/PollBuilderViewController.swift

```diff
@@ -211,51 +211,75 @@ class PollBuilderViewController: UIViewController, QuestionDelegate, PollBuilder
-                let options = mcPollBuilder.optionsDict.keys.sorted().map { mcPollBuilder.optionsDict[$0]! }
-                
-                CreateDraft(text: question!, options: options).make()
-                    .done { draft in
-                    }.catch { error in
-                        print("error: ", error)
+                var options = mcPollBuilder.options.filter { $0 != "" }
+                if options == [] {
+                    options.append("")
+                }
+                if let loadedDraft = loadedMCDraft {
+                    UpdateDraft(id: "\(loadedDraft.id)", text: question!, options: options).make()
+                        .done { draft in
+                            self.getDrafts()
+                        }.catch { error in
+                            print("error: ", error)
+                    }
+                } else {
+                    CreateDraft(text: question!, options: options).make()
+                        .done { draft in
+                            self.drafts.append(draft)
+                            self.updateDraftsCount()
+                        }.catch { error in
+                            print("error: ", error)
                     }
-                self.mcPollBuilder.clearOptionsDict()
+                }
+                self.mcPollBuilder.clearOptions()
                 self.mcPollBuilder.questionTextField.text = ""
                 self.mcPollBuilder.optionsTableView.reloadData()
             
-            } else { // FREE RESPONSE
-                //let question = frPollBuilder.questionTextField.text
-
+            case .freeResponse:
+                let question = frPollBuilder.questionTextField.text
+                if let loadedDraft = loadedFRDraft {
+                    UpdateDraft(id: "\(loadedDraft.id )", text: question!, options: []).make()
+                        .done { draft in
+                            self.getDrafts()
+                        }.catch { error in
+                            print("error: ", error)
+                    }
+                } else {
+                    CreateDraft(text: question!, options: []).make()
+                        .done { draft in
+                            self.drafts.append(draft)
+                            self.updateDraftsCount()
+                        }.catch { error in
+                            print("error: ", error)
+                    }
+                }
+                self.frPollBuilder.questionTextField.text = ""
             }
             self.updateCanDraft(false)
-            self.getDrafts()
-            }  else {
-            print("empty, drafts disabled")
         }
-        
     }
     
     @objc func startQuestion() {
-        // TODO: Start question session
-        print("start question")
-        
         // MULTIPLE CHOICE
-        if (questionType == .multipleChoice) {
-
+        guard let qType = questionType else {
```

Comment:
minor: can you do `guard let questionType = questionType`. just a bit easier to understand